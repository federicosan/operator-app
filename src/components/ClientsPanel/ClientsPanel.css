.ClientsPanel {
  background: #F7F8FA;
  box-shadow: 1px 0 1px rgba(0,0,0,0.1);
  flex-grow: 2;
  z-index: 1;
}
