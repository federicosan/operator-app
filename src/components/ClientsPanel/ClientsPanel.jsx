import React, { Component } from 'react';

import SearchBar from '../SearchBar/SearchBar.jsx';
import ClientList from '../ClientList/ClientList.jsx';
import './ClientsPanel.css';


class ClientsPanel extends Component {
  state = { query: '' }

  onQueryChange = (event) => {
    const query = event.target.value;
    this.setState({ query });
  }

  render () {
    const { query } = this.state;

    return (
      <div className="ClientsPanel">
        <SearchBar query={query} onQueryChange={this.onQueryChange} />
        <ClientList query={query} />
      </div>
    );
  }
}

export default ClientsPanel;
