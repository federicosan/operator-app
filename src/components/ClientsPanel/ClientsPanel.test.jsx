import React from 'react';

import { shallow } from 'enzyme';

import ClientsPanel from './ClientsPanel.jsx';

const store = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({ })),
};

describe('ClientsPanel', () => {
  it('matches snapshot', () => {
    const component = shallow(<ClientsPanel store={store} />);

    expect(component).toMatchSnapshot();
  });
});
