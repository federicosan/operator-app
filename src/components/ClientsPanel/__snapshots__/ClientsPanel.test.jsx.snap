exports[`ClientsPanel matches snapshot 1`] = `
<div
  className="ClientsPanel">
  <SearchBar
    onQueryChange={[Function]}
    query="" />
  <Connect(ClientList)
    query="" />
</div>
`;
